dlib/any/any_decision_function.h

```diff
@@ -79,118 +72,47 @@ namespace dlib
                 << "\n\t this: " << this
                 );
 
-            return data->evaluate(item);
+            return evaluate_func(storage.ptr, item);
         }
 
+        template<typename T>
+        bool contains() const { return storage.contains<T>();}
+        bool is_empty() const { return storage.is_empty(); }
+        void clear()          { storage.clear(); }
+        void swap (any_decision_function& item) { std::swap(storage, item.storage); }
+
```

Comment:
This doesn't swap the function pointers.
I wonder if I can simply do:
```std::swap(*this,item)```
which will do the default thing of calling move constructors and move assignment operators and do the right thing